oneargopy/Argo.py

```diff
@@ -0,0 +1,279 @@
+
+        while (not success) and (iterations < self.download_settings.max_attempts):
+            # Try both hosts (perfered one is listed first in download settings)
+            for host in self.source_settings.hosts:
+
+                url = "".join([host, file_name, ".gz"])
+
+                if self.download_settings.verbose: print(f'Downloading {file_name} from {url}...')
+
+                try:
+                    with requests.get(url, stream=True) as r:
+                        r.raise_for_status()
+                        with open(gz_save_path, 'wb') as f:
+                            r.raw.decode_content = True
+                            shutil.copyfileobj(r.raw, f)
+
+                    if self.download_settings.verbose: print(f'Unzipping {file_name}.gz...')
+                    with gzip.open(gz_save_path, 'rb') as gz_file:
+                        with open(txt_save_path, 'wb') as txt_file:
+                            shutil.copyfileobj(gz_file, txt_file)
+                    
+                    success = True
+                    if self.download_settings.verbose: print(f'Success!')
+
+                    # Remove extraneous .gz file
+                    gz_save_path.unlink()
+                    
+                    # Exit the loop if download is successful so we don't try additional
+                    # sources for no reason.
+                    break 
+
+                except requests.RequestException as e:
+                    print(f'Error encountered: {e}. Trying next host...')
+            
+            # Increment Iterations
+            iterations += 1
+
+        # If ultimately nothing could be downloaded
+        if not success: 
+            if update_status:
+                print(f'WARNING: Update for {file_name} failed, you are working with out of date data.')
+            else:
+             raise Exception(f'Download failed! {file_name} could not be downloaded at this time.')
+        
+
+    def __load_synthetic_dataframe(self) -> pd:
+        """ A function to load an index file into a data frame for easier reference.
+
+            :param: file_name : str - The name of the file that we would like
+                to read into a dataframe.
+
+            NOTE: the header is 8 here because there are 8 lines in both index fiels
+                devoted to header information.
+            NOTE: R: raw data, A: adjusted mode (real-time adjusted), D: delayed mode quality controlled
+
+        """
+        file_name = "argo_synthetic-profile_index.txt"
+        file_path = Path.joinpath(self.download_settings.base_dir, 'Index', file_name)
+        synthetic_index = pd.read_csv(file_path, delimiter=',', header=8, parse_dates=['date','date_update'], 
+                                date_format='%Y%m%d%H%M%S')
+        
+        # Parsing out variables in first column file
+        dacs = synthetic_index['file'].str.split('/').str[0]
+        synthetic_index.insert(0, "dacs", dacs, True)
```

Comment:
Why should we allow duplicates of these columns?

Reply:
I'm not 100% sure what you mean? Do you mean how the 'file' column is still in both dataframes even after extracting the 'dacs', 'wmoid', and 'profile' information? 

I left 'file' in but separated the information because with each variable in it's own column searching for specific information becomes a lot easier without having to write the split logic each time. I left the 'file' column because I wasn't sure if we used the full file path later on when downloading .nc files or not.